Kitodo/src/main/java/org/kitodo/production/metadata/copier/DataCopyrule.java

```diff
@@ -92,44 +47,46 @@ public static DataCopyrule createFor(List<String> arguments) throws Configuratio
-     * required by the rule to work as expected. If it returns 0, the
-     * setObjects() method will not be called.
-     *
-     * @return the maximal number of objects required by the rule
-     */
-    protected abstract int getMaxObjects();
+    public void apply(CopierData data) {
+        Workpiece workpiece = data.getDigitalDocument().getWorkpiece();
+        List<IncludedStructuralElement> allIncludedStructuralElements = workpiece.getAllIncludedStructuralElements();
 
-    /**
-     * The function getMinObject must return the minimal number of objects
-     * required by the rule to work as expected.
-     *
-     * @return the minimal number of objects required by the rule
-     */
-    protected abstract int getMinObjects();
+        for (IncludedStructuralElement child : allIncludedStructuralElements) {
+            Collection<Metadata> metadata = child.getMetadata();
+            MdSec domain = null;
+            // TODO: should be replaced by further methodes
+            // for (Metadata metadatum : metadata) {
+            // domain = metadatum.getDomain();
+            //
+            // if (metadatum instanceof MetadataEntry) {
+            // if (metadatum.getKey().equals(this.command.get(0))) {
+            // ((MetadataEntry) metadatum).setValue(this.command.get(2));
+            // break;
+            // }
+            // }
+            // if (metadatum instanceof MetadataGroup) {
+            // Collection<Metadata> group = ((MetadataGroup) metadatum).getGroup();
+            // if (metadatum.getKey().equals(this.command.get(0))) {
+            //
+            // }
+            // for (Metadata groupelement : group) {
+            // }
+            // }
+            // }
+            MetadataEntry metadataEntry = new MetadataEntry();
+            metadataEntry.setKey(command.get(0));
+            metadataEntry.setValue(command.get(2));
+            metadataEntry.setDomain(domain);
+            child.getMetadata().add(metadataEntry);
 
-    /**
-     * The method is called to pass the rule its objects. The list
-     * passed is reliable to the restrictions defined by getMinObjects() and
-     * getMaxObjects().
-     *
-     * @param objects
-     *            a list of objects to be used by the rule
-     * @throws ConfigurationException
-     *             may be thrown if one of the objects cannot be processed
-     */
-    protected abstract void setObjects(List<String> objects) throws ConfigurationException;
+            try (OutputStream out = ServiceManager.getFileService()
+                    .write(ServiceManager.getFileService().getMetadataFilePath(data.getProcess()))) {
+                ServiceManager.getMetsService().save(workpiece, out);
+                ServiceManager.getProcessService().saveToIndex(data.getProcess(), false);
+            } catch (IOException | CustomResponseException | DataException e) {
+                System.out.println("log me");
```

Comment:
please use: `Helper.setErrorMessage(e.getMessage(), logger, e)`

Reply:
I'm going to remove this class, and as PRs are stacking up please be generous to spare me conflicts. :-)